analyses/molecular-subtyping-ATRT/01-ATRT-molecular-subtyping-data-prep.Rmd

```diff
@@ -0,0 +1,391 @@
+myc_genes <- c("MYC", "HOTAIR", "HOX")
+
+# Filter expression data for target overexpressed genes
+stranded_expression_wide_df <- stranded_expression %>%
+  as.data.frame() %>%
+  dplyr::mutate(gene_id = gsub(".*\\_", "", gene_id)) %>% # Trim the gene ids to include only the gene symbol
+  dplyr::filter(gene_id %in% tyr_genes | gene_id %in% shh_genes | gene_id %in% myc_genes) %>%
+  tibble::column_to_rownames("gene_id")
+
+# Log2 transformation
+norm_expression <- log2(stranded_expression_wide_df + 1)
+
+# normData mean and sd
+norm_expression_means <- rowMeans(norm_expression, na.rm = TRUE)
+norm_expression_sd <- apply(norm_expression, 1, sd, na.rm = TRUE)
+
+# Subtract mean
+expression_zscored <- sweep(stranded_expression_wide_df, 1, norm_expression_means, FUN = "-")
+
+# Divide by SD remove NAs and Inf values from zscore for genes with 0 in normData
+expression_zscored <- sweep(expression_zscored, 1, norm_expression_sd, FUN = "/") %>% 
+  dplyr::na_if(Inf) %>% 
+  na.omit()
+
+# Transpose
+long_stranded_expression <- t(expression_zscored)
+
+# Merge metadata with expression data 
+expression_metadata <- long_stranded_expression %>%
+  as.data.frame() %>%
+  tibble::rownames_to_column("Kids_First_Biospecimen_ID") %>%
+  dplyr::left_join(select_metadata, by = "Kids_First_Biospecimen_ID")
+
+# Display `expression_metadata`
+expression_metadata %>%
+  head(n = 15) %>%
+  regulartable() %>%
+  fontsize(size = 12, part = "all")
+
+# Join expression data with metadata filtered for `ATRT`
+atrt_expression_df <- atrt_df %>%
+  dplyr::left_join(expression_metadata,
+                    by = "sample_id")
+```
+
+### CN data
+
+```{r}
+# Collapse target overexpressed gene vectors
+tyr_genes <- paste(tyr_genes, collapse = "|")
+shh_genes <- paste(shh_genes, collapse = "|")
+myc_genes <- paste(myc_genes, collapse = "|")
+
+# Join focal CN data with metadata
+cn_metadata <- cn_df %>%
+  dplyr::left_join(metadata,
+                   by = c("biospecimen_id" = "Kids_First_Biospecimen_ID")) %>%
+  dplyr::select(gene_symbol,
+                sample_id,
+                Kids_First_Participant_ID,
+                biospecimen_id,
+                status) %>%
+  dplyr::filter(sample_id %in% atrt_expression_df$sample_id) %>%
+  dplyr::mutate(
+    SMARCB1_focal_status = dplyr::case_when(gene_symbol == "SMARCB1" ~ status,
+                                            TRUE ~ "neutral"),
+    SMARCA4_focal_status = dplyr::case_when(gene_symbol == "SMARCA4" ~ status,
+                                            TRUE ~ "neutral"),
+    Overexpressed_gene_sets = dplyr::case_when(
+      stringr::str_detect(gene_symbol, tyr_genes) &
+        status != "loss" ~ "TYR_genes",
+      stringr::str_detect(gene_symbol, shh_genes) &
+        status != "loss" ~ "SHH_genes",
+      stringr::str_detect(gene_symbol, myc_genes) &
+        status != "loss" ~ "MYC_genes",
+      TRUE ~ "NA"
+    )
+  ) %>%
+  dplyr::select(-gene_symbol) %>%
```

Comment:
Probably drop `status` here too?

Reply:
If included `status` is for display, I'd probably leave `gene_symbol` here and take it out at line 279.